src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -82,31 +85,36 @@ def isEnabled(self, service):
         return enabled
 
     @private
-    def tryConnect(self, host, port):
+    def tryConnect(self, host, port, fnldap):
         max_tries = 3
         connected = False
 
-        timeout = _fs().middlewared.plugins.service_monitor.socket_timeout
+        if self.name == 'activedirectory':
+            host_list = []
 
-        for i in range(0, max_tries):
-            # XXX What about UDP?
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(timeout)
-
-            try:
-                s.connect((host, port))
-                connected = True
+            for i in range(0, max_tries):
+                # Make max_tries attempts to get SRV records from DNS
+                host_list = fnldap.get_ldap_servers(host)
+                if host_list:
+                    break
+                else:
+                    self.logger.debug("[ServiceMonitorThread] Attempt %d to query SRV records failed " % (i))
 
-            except Exception as e:
-                self.logger.debug("[ServiceMonitorThread] Cannot connect: %s:%d with error: %s" % (host, port, e))
-                connected = False
+            if not host_list:
+                self.logger.debug("[ServiceMonitorThread] Query for SRV records for %s failed" % (host))
+                return False
 
-            finally:
-                s.settimeout(None)
-                s.close()
+        else:
+            self.logger.debug("[ServiceMonitorThread] no monitoring has been written for %s " % self.name)
+            return False
 
-            if connected:
-                break
+        for h in host_list:
```

Comment:
It's a bit hard to see it here, but looks like `host_list` you are referencing here is declared, if any, in another scope. I actually commented about this to you before. 
It's declared on an inner scope regarding this one. Also, if `if self.name != 'activedirectory':` - it's not declared at all. Move `host_list = []` outside of the `if`.